syntax = "proto3";

package playwright;

service PlaywrightScanner {
  rpc Scan(ScanRequest) returns (stream ScanResult);
  rpc HealthCheck(HealthRequest) returns (HealthResponse);
}

message ScanRequest {
  string url = 1;
  int32 max_depth = 2;
}

message ScanResult {
  oneof result {
    KatanaOutput data = 1;
    ScanError error = 2;
  }
}

message KatanaOutput {
  Request request = 1;
  Response response = 2;
  double timestamp = 3;
}

message Request {
  string method = 1;
  string endpoint = 2;
  map<string, string> headers = 3;
  optional string body = 4;
  string raw = 5;
}

message Response {
  int32 status_code = 1;
  map<string, string> headers = 2;
}

message ScanError {
  string message = 1;
}

message HealthRequest {}

message HealthResponse {
  string status = 1;
  string service = 2;
}
